File: code-companion/src/types/providers.ts
/**
 * Provider 配置相关类型定义
 */

/**
 * Claude 引擎当前配置
 */
export interface ClaudeCurrentConfig {
  env?: Record<string, string>;
  apiKeyHelper?: string;
}

/**
 * Codex 引擎当前配置
 */
export interface CodexCurrentConfig {
  auth?: Record<string, any>;
  config?: Record<string, any>;
}

/**
 * Gemini 引擎当前配置
 */
export interface GeminiCurrentConfig {
  env?: Record<string, string>;
  settings?: Record<string, any>;
}

/**
 * Provider 当前配置联合类型
 */
export type ProviderCurrentConfig = ClaudeCurrentConfig | CodexCurrentConfig | GeminiCurrentConfig;

/**
 * Provider 配置响应
 */
export interface ProviderConfigResponse {
  engine: string;
  config: ProviderCurrentConfig;
}

/**
 * Provider 配置更新请求
 */
export interface ProviderConfigUpdateRequest {
  apiKey?: string | null;
  baseUrl?: string | null;
  enabled?: boolean | null;
  model?: string | null;
  temperature?: number | null;
}

/**
 * Provider 预设
 */
export interface ProviderPreset {
  id: string;
  name: string;
  description?: string;
  config: Record<string, any>;
}

/**
 * Provider 预设响应
 */
export interface ProviderPresetResponse {
  id: string;
  name: string;
  description?: string;
  config: Record<string, any>;
}

/**
 * Provider 预设创建请求
 */
export interface ProviderPresetCreateRequest {
  id: string;
  name: string;
  description?: string;
  config: Record<string, any>;
}

File: backend/app/schemas/providers.py
"""
Provider 配置管理相关的 Pydantic 模型
"""
from typing import Any, Dict, Optional
from pydantic import BaseModel, Field


class ProviderConfigResponse(BaseModel):
    """Provider 配置响应"""
    engine: str = Field(description="引擎名称")
    config: Dict[str, Any] = Field(description="配置数据")


class ProviderConfigUpdateRequest(BaseModel):
    """Provider 配置更新请求"""
    apiKey: Optional[str] = Field(default=None, description="API Key")
    baseUrl: Optional[str] = Field(default=None, description="API Base URL")
    enabled: Optional[bool] = Field(default=None, description="是否启用")
    model: Optional[str] = Field(default=None, description="默认模型")
    temperature: Optional[float] = Field(default=None, description="温度参数")


class ProviderPresetResponse(BaseModel):
    """Provider 预设响应"""
    id: str = Field(description="预设 ID")
    name: str = Field(description="预设名称")
    description: Optional[str] = Field(default=None, description="预设描述")
    config: Dict[str, Any] = Field(description="完整配置数据")


class ProviderPresetCreateRequest(BaseModel):
    """Provider 预设创建/更新请求"""
    id: str = Field(description="预设 ID")
    name: str = Field(description="预设名称")
    description: Optional[str] = Field(default=None, description="预设描述")
    config: Dict[str, Any] = Field(description="完整配置数据")
